### DISCUSSION TITLE
When encrypted data is found, and no privateKey, then a config error should be raised

### DISCUSSION THREAD

**Discussion by justo-mend** [renovatebot/renovate#33224] [Suggest an Idea]
### Tell us more.

Currently, when encrypted data is found but there is no privateKey, the problem is reported in the Dependency Dashboard only.

It would be better if this was raised as a Config error issue.

**Comment by rarkins** [comment]
Agreed. Using `encrypted` objects in config where it's not supported by a global `privateKey` is a config error and should raise an issue/abort the job

**Comment by justo-mend** [comment]
When the CLI detects that that it was called from Mend-hosted servers (ie. env var `MEND_HOSTED=true`), it should replace the issue body with the following text:

```
Mend-hosted Renovate Apps no longer support the use of encrypted secrets in Renovate file config (ie. renovate.json).
Please migrate all secrets to the Developer Portal using the web UI available at https://developer.mend.io/

Refer to migration documents here: https://docs.renovatebot.com/mend-hosted/migrating-secrets/
```

**Reply by rarkins** [reply]
@justo-mend you need to specify the env

**Reply by justo-mend** [reply]
Let's look for the env var `MEND_HOSTED` to be "true".

**Comment by RahulGautamSingh** [comment]
If you set `RENOVATE_X_ENCRYPTED_STRICT=true` in the env. Bot will raise an issue.

Extra addition needed is the custom message for `MEND_HOSTED=true`.

Or, do we want to raise an issue regardless `RENOVATE_X_ENCRYPTED_STRICT` is set or not?

**Reply by rarkins** [reply]
For now, throw error/raise issue only if the RENOVATE_X is defined
